nengo_spa/tests/test_semantic_pointer.py

```diff
@@ -317,9 +318,136 @@ def test_name():
+
+        # check that (sqrt(x))**2 == x
+        assert np.allclose(((x ** 0.5) ** 2).v, x.v)
+
+        # multiplication will add the exponents
+        assert np.allclose((x ** 2 * x).v, (x ** 3).v)
+        assert np.allclose((x ** 2.5 * x ** 1.5).v, (x * x * x * x).v)
+
+        # check this more generally
+        a, b, c = 10 * rng.rand(3) - 5  # [-5, 5]
+        assert np.allclose((x ** a * x ** b * x ** c).v, (x ** (a + b + c)).v)
+
+        # counter-intuitively, the property:
+        #     (x**a)**b == x**(a*b)
+        # does not hold in general -- e.g. (x**2)**(0.5) can be +/- x
+        # that is because sqrt is a multi-valued function
+        # assert np.allclose(((x**a)**b).v, (x**(a*b)).v)
+
+
+@pytest.mark.parametrize("d", [1, 4, 9, 16, 129, 512])
+def test_fractional_binding_unitary(d, rng):
+    # tests a number of properties of fractional binding with nondegenerate
+    # and unitary vectors
+
+    gen = UnitaryVectors(d, algebra=HrrAlgebra(), rng=rng)
+    for _ in range(100):
+        x = SemanticPointer(next(gen)).nondegenerate()
+
+        # check that making it nondegenerate or unitary again
+        # does not change the vector
+        assert np.allclose((x.nondegenerate()).v, x.v)
+        assert np.allclose((x.unitary()).v, x.v)
+
+        # 1/x is the inverse (since unitary)
+        assert np.allclose((x ** (-1)).v, (~x).v)
+
+        # x**0 gives the identity (since unitary)
+        assert np.allclose((x ** 0).v, Identity(d).v)
+
+        # check that it's still unitary after fractional binding
+        sqrt_x = x ** 0.5
+        x_check = sqrt_x * sqrt_x
+
+        assert np.allclose(np.linalg.norm(sqrt_x.v), 1)
+        assert np.allclose(np.linalg.norm(x_check.v), 1)
+        assert np.allclose(x_check.v, x.v)
+
+
+def test_degenerate():
+    # tests that an error occurs if throwing away imag components
+    # this error occurs regardless of whether it's unitary
+    x = SemanticPointer([1, 2, 3, 4, 5, 6]).unitary()
+    with pytest.raises(ValueError, match="nondegenerate"):
+        x ** 0.5
+
+
+def test_fractional_binding_natural():
+    # tests fractional binding with natural exponents
+    x = SemanticPointer([1, 2, 3, 4, 5, 6])
+
+    assert np.allclose((x ** 1).v, x.v)
+    assert np.allclose((x ** 2).v, (x * x).v)
+    assert np.allclose((x ** 4).v, (x * x * x * x).v)
+
+    # check that x is degenerate (i.e., not nondegenerate)
+    assert not np.allclose(x.nondegenerate().v, x.v)
+
+
+@pytest.mark.parametrize("d", [1, 4, 9, 144])
+def test_fractional_binding_vtb(d, rng):
+    pytest.importorskip("scipy")
+
+    algebra = VtbAlgebra()
+    v = next(UnitLengthVectors(d, rng=rng))
+    x = SemanticPointer(v, algebra=algebra)
+
+    # note: doesn't quite behave with ideal properties
+    assert np.allclose((x ** 0).v, algebra.identity_element(d))
+    assert np.allclose((x ** 1).v, (~x).v)
+    assert np.allclose((x ** 2).v, (~x * x).v)
+    assert np.allclose((x ** 3).v, (((~x) * x) * x).v)
```

Comment:
Why does `x**1` give `~x`?

Reply:
Had to do with how I implemented fractional binding for VTB. Because it's repeated matrix multiplication, but a transpose is needed on one side (related to the non-commutativity). Sounds like @bjkomer has an alternative solution (https://github.com/nengo/nengo-spa/pull/243#discussion_r421039596).

Reply:
Probably it was a mistake to define `B(x, y) = V_y x` instead of `B(x, y) = V_x y`. But well ... it seems to me that after doing the fractional binding, the identity is bound on the wrong side (`B(fractional_bound_vector, identity)` instead of `B(identity, fractional_bound_vector`)). Due to the definition, it might not be possible to do the binding on correct side directly, but applying the "swapping matrix" (Eq. 2.23 in the original VTB paper, there is an implementation in the algbra too) should do the trick. It's actually equivalent to the inversion matrix which explains the results above.